WireUI/Sources/WireUIFoundation/Main/MainCoordinator.swift

```diff
@@ -0,0 +1,364 @@
+    SelfProfileBuilder: MainCoordinatorInjectingViewControllerBuilder
+
+>: NSObject, MainCoordinatorProtocol, UISplitViewControllerDelegate, UITabBarControllerDelegate where
+
+    SplitViewController.Sidebar: MainSidebarProtocol,
+    SplitViewController.ConversationList == TabBarController.ConversationList,
+    SplitViewController.Archive == TabBarController.Archive,
+    SplitViewController.Settings == TabBarController.Settings
+{
+    // swiftlint:enable opening_brace
+
+    // MARK: - Private Properties
+
+    private weak var splitViewController: SplitViewController!
+    private weak var tabBarController: TabBarController!
+
+    private let newConversationBuilder: NewConversationBuilder
+    private weak var newConversation: UIViewController?
+
+    private var selfProfileBuilder: SelfProfileBuilder
+    private weak var selfProfile: UIViewController?
+
+    private var mainSplitViewState: MainSplitViewState = .expanded
+
+    // MARK: - Private Helpers
+
+    private var sidebar: SplitViewController.Sidebar! {
+        splitViewController.sidebar
+    }
+
+    private var conversationList: TabBarController.ConversationList! {
+        switch mainSplitViewState {
+        case .collapsed: tabBarController.conversations?.conversationList
+        case .expanded: splitViewController.conversationList
+        }
+    }
+
+    private var archive: TabBarController.Archive! {
+        switch mainSplitViewState {
+        case .collapsed: tabBarController.archive
+        case .expanded: splitViewController.archive
+        }
+    }
+
+    private var settings: TabBarController.Settings! {
+        switch mainSplitViewState {
+        case .collapsed: tabBarController.settings
+        case .expanded: splitViewController.settings
+        }
+    }
+
+    // MARK: - Life Cycle
+
+    public init(
+        mainSplitViewController: SplitViewController,
+        mainTabBarController: TabBarController,
+        newConversationBuilder: NewConversationBuilder,
+        selfProfileBuilder: SelfProfileBuilder
+    ) {
+        guard
+            mainSplitViewController.conversationList != nil,
+            mainTabBarController.archive != nil,
+            mainTabBarController.settings != nil
+        else { fatalError("invalid state of container view controllers") }
+
+        splitViewController = mainSplitViewController
+        tabBarController = mainTabBarController
+        self.newConversationBuilder = newConversationBuilder
+        self.selfProfileBuilder = selfProfileBuilder
+    }
+
+    // MARK: - Public Methods
+
+    public func showConversationList(conversationFilter: (some MainConversationFilterRepresentable)?) {
+        defer {
+            // switch to the conversation list tab
+            tabBarController.selectedContent = .conversations
+
+            // apply the filter to the conversation list
+            let mainConversationFilter = conversationFilter?.map()
```

Comment:
Optional: Looking at this via GitHub diff, it is confusing what `map` does here. Consider renaming to something more explicit like `mainConversationFilter()` or `mappedToMainConversationFilter()`

Reply:
Done, thanks.